internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -47,6 +48,206 @@ func New(sess *session.Session) CloudFormation {
 	}
 }
 
+// CreateProjectResources sets up everything required for our project-wide resources.
+// These resources include things that are regional, rather than scoped to a particular
+// environment, such as ECR Repos, CodePipeline KMS keys & S3 buckets.
+// We deploy project resources through StackSets - that way we can have one
+// template that we update and all regional stacks are updated.
+func (cf CloudFormation) CreateProjectResources(project *archer.Project) error {
+	projectConfig := newProjectStackConfig(project, cf.box)
+
+	// First deploy the project roles needed by StackSets. These roles
+	// allow the stack set to set up our regional stacks.
+	if err := cf.deploy(projectConfig); err == nil {
+		_, err := cf.waitForStackCreation(projectConfig)
+		if err != nil {
+			return err
+		}
+	} else {
+		// If the stack already exists - we can move on
+		// to creating the StackSet.
+		var alreadyExists *ErrStackAlreadyExists
+		if !errors.As(err, &alreadyExists) {
+			return err
+		}
+	}
+
+	blankProjectTemplate, err := projectConfig.ResourceTemplate(&projectResourcesConfig{
+		Project: projectConfig.Project.Name,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	_, err = cf.client.CreateStackSet(&cloudformation.CreateStackSetInput{
+		Description:           aws.String(projectConfig.StackSetDescription()),
+		StackSetName:          aws.String(projectConfig.StackSetName()),
+		TemplateBody:          aws.String(blankProjectTemplate),
+		ExecutionRoleName:     aws.String(projectConfig.StackSetExecutionRoleName()),
+		AdministrationRoleARN: aws.String(projectConfig.StackSetAdminRoleARN()),
+		Tags:                  projectConfig.Tags(),
+	})
+
+	if err != nil && !stackSetExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+// UpdateProject updates Project resources in all regions. If a new app or env is added
+func (cf CloudFormation) UpdateProject(project *archer.Project, newApps []*archer.Application, newEnvs []*archer.Environment) error {
+	projectConfig := newProjectStackConfig(project, cf.box)
+
+	if err := cf.updateStackSet(projectConfig, newApps, newEnvs); err != nil {
+		return err
+	}
+	return cf.addNewStackInstances(projectConfig, newEnvs)
```

Comment:
dumb question: When we add a new stack instances is the existing template trickled down to the new instance?

I'm asking to understand if the order of these two methods should be swapped or not

Reply:
Yeap. When we update the stack set - any existing stack instances are updated. THEN, when when we add a new stack instance, it gets deployed with the latest stack template.